Deep link support

estib-vega opened the pull request:
Add support for deeplinking the application.
This will enable opening the application from the web or from the terminal

Related to: https://github.com/gitbutlerapp/gitbutler/issues/11025

Byron commented on crates/but/src/gui.rs:
```diff
@@ -1,43 +1,74 @@
+
+        #[cfg(target_os = "macos")]
+        {
+            std::process::Command::new("open")
```
If system URL associations are used, then I think the `open` crate is a more portable choice here.
Disclaimer: It's a crate I maintain, and it's already available via `open.workspace = true`.

estib-vega replied:
Uhhh that's cool

estib-vega committed:
but: GUI command

Update the command so that it opens the GUI through the deeplinks

estib-vega committed:
Deep link support

Add support for deeplinking the application.
This will enable opening the application from the web or from the terminal

estib-vega committed:
Project: Add project with best effort matching

Try to match the given path to all existing projects to determine whether it is a path inside a project repository.
Otherwise, handle adding the project as usual.

estib-vega committed:
Deep Link: Handle opening a project

Open or add a project from a deeplink
